### PR TITLE
CAMEL-15664: Automatically wrap secret properites with RAW when computing the URI

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by davsclaus** [comment]
@lburgazzoli I wonder if instead of chaning from set to map that would be much bigger in memory due to 99,8% of their values are just false. But instead add a new API that returns the propertyNames for the options that are secret. Then we can keep the existing as a Set.

**Comment by lburgazzoli** [comment]
ah yeah that could be better, I'll rework the PR next week

**Comment by lburgazzoli** [comment]
@davsclaus done, can you have a look ?

**Review by oscerd — APPROVED** [review]

**Comment by davsclaus** [comment]
Why are we using a TreeMap now? If the keys should be case insensitive, then dont you need to provide a comparator that does that? 

And if the keys should be sorted ??? then we could also make the source code pre-sort so the insert is faster (no need to reorder while adding).

File: core/camel-support/src/main/java/org/apache/camel/support/component/EndpointUriFactorySupport.java [hunk 1]
```diff
@@ -78,6 +79,20 @@ protected String buildQueryParameters(String uri, Map<String, Object> parameters
             throws URISyntaxException {
         // we want sorted parameters
         Map<String, Object> map = new TreeMap<>(parameters);
+        map.replaceAll(new BiFunction<String, Object, Object>() {
```
Comment (davsclaus) [inline core/camel-support/src/main/java/org/apache/camel/support/component/EndpointUriFactorySupport.java:82]: Isnt there a way to do this faster, eg now you replace all values? What if there are no secrets at all? Wont it be faster just to walk on the secret property names, and see if they are in the parameters, and then only replace those.

**Comment by lburgazzoli** [comment]
> Why are we using a TreeMap now? If the keys should be case insensitive, then dont you need to provide a comparator that does that?
> 

I don't think I have changed it, looks like the original code [uses](https://github.com/apache/camel/blob/95a3587ebeda05e956487b78d0163db2b7fcae67/core/camel-support/src/main/java/org/apache/camel/support/component/EndpointUriFactorySupport.java#L80) TreeMap already, should I chnage it ?

**Comment by lburgazzoli** [comment]
@davsclaus should be better now

**Comment by davsclaus** [comment]
Yeah its better, the initial size of the set was lost, I think it should be added back

**Comment by lburgazzoli** [comment]
yep noticed it but I ran out of allowed time to work over the week end :)
I was also wondering if we can replace them (the sets) with plain arrays

**Comment by davsclaus** [comment]
the sets are used to have fast lookup as its used in toD during Camel routing (for some components like http based). If we really really wanted, we could make those SendDynamic (toD optimized) a way of marking which options they "use" and then have the source code generator generate a special set for those, and then an array in general. 

But from what you need in ckc then using array/set doesnt really matter. An array may take up less space in memory, than a hashset.

**Comment by lburgazzoli** [comment]
> the sets are used to have fast lookup as its used in toD during Camel routing (for some components like http based). If we really really wanted, we could make those SendDynamic (toD optimized) a way of marking which options they "use" and then have the source code generator generate a special set for those, and then an array in general.
> 

Right, I've discovered it this morning so no need to change to array.

**Comment by lburgazzoli** [comment]
@davsclaus I should have cleaned up all my rubbish now

**Review by davsclaus — APPROVED** [review]

**Pull request merged**